uitools/examples/cpp_quick/src/BaseDemo.cpp

```diff
@@ -0,0 +1,132 @@
+#include "BaseDemo.h"
+
+#include <Map.h>
+#include <MapQuickView.h>
+#include <MapViewTypes.h>
+#include <Scene.h>
+#include <SceneQuickView.h>
+#include <stdexcept>
+
+using namespace Esri::ArcGISRuntime;
+
+BaseDemo::BaseDemo(QObject* parent) :
+  QObject(parent),
+  m_geoView(nullptr)
+{
+}
+
+BaseDemo::~BaseDemo()
+{
+}
+
+Esri::ArcGISRuntime::GeoView* BaseDemo::geoView() const
+{
+  if (auto mapView = qobject_cast<MapQuickView*>(m_geoView))
+  {
+    return mapView;
+  }
+  else if (auto sceneView = qobject_cast<SceneQuickView*>(m_geoView))
+  {
+    return sceneView;
+  }
+  return nullptr;
+}
+
+QObject* BaseDemo::geoView_() const
+{
+  return m_geoView;
+}
+
+void BaseDemo::setGeoView(Esri::ArcGISRuntime::GeoView* geoView)
+{
+  if (geoView == nullptr)
+  {
+    setGeoView_(nullptr);
+  }
+  else
+  {
+    if (geoView->geoViewType() == GeoViewType::MapView)
+    {
+      setGeoView_(static_cast<MapQuickView*>(geoView));
+    }
+    else if (geoView->geoViewType() == GeoViewType::SceneView)
+    {
+      setGeoView_(static_cast<SceneQuickView*>(geoView));
+    }
+    else
+    {
+      throw std::runtime_error("Unknown type of GeoView");
```

Comment:
🔧  perhaps `Q_ASSERT_X`? Uncommon to throw exceptions in public-facing Runtime Qt code. Qt is exception agnostic.

Reply:
Good point, I've replaced with `qFatal` as that is what `Q_ASSERT` uses under the hood when a failure occurs.